ext/symengine/ruby_ntheory.c

```diff
@@ -0,0 +1,105 @@
+#include "ruby_ntheory.h"
+
+VALUE cntheory_onearg(void (*cwfunc_ptr)(basic_struct*, const basic_struct*), VALUE operand1) {
+    basic_struct *cresult;
+    VALUE result;
+
+    basic cbasic_operand1;
+    basic_new_stack(cbasic_operand1);
+    sympify(operand1, cbasic_operand1);
+
+    cresult = basic_new_heap();
+    cwfunc_ptr(cresult, cbasic_operand1);
+    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL , cbasic_free_heap, cresult);
+    basic_free_stack(cbasic_operand1);
+
+    return result;
+}
+
+VALUE cntheory_twoarg(void (*cwfunc_ptr)(basic_struct*, const basic_struct*, const basic_struct*), VALUE operand1, VALUE operand2) {
+    basic_struct *cresult;
+    VALUE result;
+
+    basic cbasic_operand1;
+    basic_new_stack(cbasic_operand1);
+    sympify(operand1, cbasic_operand1);
+
+    basic cbasic_operand2;
+    basic_new_stack(cbasic_operand2);
+    sympify(operand2, cbasic_operand2);
+
+    cresult = basic_new_heap();
+    cwfunc_ptr(cresult, cbasic_operand1, cbasic_operand2);
+    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL , cbasic_free_heap, cresult);
+    basic_free_stack(cbasic_operand1);
+    basic_free_stack(cbasic_operand2);
+
+    return result;
+}
+
+VALUE cntheory_nextprime(VALUE self, VALUE operand1) {
+    return cntheory_onearg(ntheory_nextprime, operand1);
+}
+
+VALUE cntheory_gcd(VALUE self, VALUE operand1, VALUE operand2) {
+    return cntheory_twoarg(ntheory_gcd, operand1, operand2);
+}
+
+VALUE cntheory_lcm(VALUE self, VALUE operand1, VALUE operand2) {
+    return cntheory_twoarg(ntheory_lcm, operand1, operand2);
+}
+
+VALUE cntheory_mod(VALUE self, VALUE operand1, VALUE operand2) {
+    return cntheory_twoarg(ntheory_mod, operand1, operand2);
+}
+
+VALUE cntheory_quotient(VALUE self, VALUE operand1, VALUE operand2) {
+    return cntheory_twoarg(ntheory_quotient, operand1, operand2);
+}
+
+VALUE cntheory_fibonacci(VALUE self, VALUE operand1){
+    basic_struct *cresult;
+    VALUE result;
+
+    unsigned long cbasic_operand1;
+    cbasic_operand1 = NUM2ULONG(operand1);
```

Comment:
I figured our half way on how to do this:

so I added a new method to the 'Integer' class
'rb_define_method(c_integer, "to_int", cinteger_to_int, 0);'

From which I should return a VALUE pointer pointing to a Ruby `Bignum` or a `Fixnum`. This can be done by converting a C int by using `INT2NUM` macro.

The step I'm stuck at is on extracting a C int from the SymEngine Integer object. @abinashmeher999 can you point me to somewhere I can read more about the Ruby Wrapper's SymEngine Integer's structure?